sd/etcd/registrar.go

```diff
@@ -43,6 +67,22 @@ func (r *Registrar) Register() {
 	} else {
 		r.logger.Log("action", "register")
 	}
+	if r.service.TTL == nil {
+		return
+	}
```

Comment:
This says "bail early if there's no configured TTL" — but I think it's much more clear if the logic is inverted, i.e. "if there _is_ a configured TTL, create the quit chan and launch the maintenance loop".